### ISSUE TITLE
Do we have any plans to support OIDC?

### ISSUE THREAD

**Issue by saltbo** [envoyproxy/envoy#21982]
*Title*: *Do we have any plans to support OIDC?*

*Description*:
> OAuth2 config is tedious, if support oidc, we only need config the discover api to implement the all oauth2 feature

[optional *Relevant Links*:]
- https://developers.google.com/identity/protocols/oauth2/openid-connect#discovery
- https://openid.net/specs/openid-connect-core-1_0.html

**Comment by wbpcode** [comment]
cc @snowp

**Comment by snowp** [comment]
I'm not aware of any plans, so it will likely require someone to do the work. PRs welcome!

cc @rgs1

**Comment by saltbo** [comment]
I found a plugin https://github.com/dgn/oidc-filter

**Comment by NabeelShahid7** [comment]
I would like to work on this enhancement.. Can I please get some kind of an approval from any contributor of this repository?
About Me:
I am a student currently pursuing his Post Graduate in Computer Science. I have been assigned a course component which involves contributing to any open source project. I am liking this repository and would like to start from scratch to try getting this issue fixed. @snowp

**Comment by loewenstein** [comment]
Looking at https://github.com/envoyproxy/gateway/issues/881, Envoy Gateway seems to consider this topic "done".

On the other hand side, given the title of this issue is very broad I'd be interested if there are any ideas, plans or even just appetite for looking into additional specs like [OIDC Session Management](https://openid.net/specs/openid-connect-session-1_0.html), [OIDC Front-Channel Logout](https://openid.net/specs/openid-connect-frontchannel-1_0.html) and [OIDC Back-Channel Logout](https://openid.net/specs/openid-connect-backchannel-1_0.html)?

**Comment by ngorantla-equinix** [comment]
anyone working on this implementation?

**Comment by loewenstein** [comment]
@ngorantla-equinix I don't think so. I would first of all be interested to hear if there is appetite for "more of OIDC"?

**Comment by antonengelhardt** [comment]
I worked on this, and I am very happy to receive any feedback. https://github.com/antonengelhardt/wasm-oidc-plugin
